content/developer/reference/guidelines.rst

```diff
@@ -999,17 +999,88 @@ More precise JS guidelines are detailed in the `github wiki  <https://github.com
 You may also have a look at existing API in Javascript by looking Javascript
 References.
 
-CSS coding guidelines
----------------------
+Template guidelines
+-------------------
+
+Structure
+~~~~~~~~~
+
+.. note:: These rules are in place because the general structure of the template is considered more important than the details of each line.
+
+- Each block element should be on its own line.
+- Small inline/style/text elements can be placed either on the same line as their parent or on their own line (depending on their importance and size).
+- Indentation of 4 spaces is applied between a parent and its children elements. The closing tag of a parent should always be at the same indentation level as its opening tag.
+- It is preferable to use instructions (such as `t-if`) directly on the element on which they apply (as opposed to having them on a separate `<t>` element).
+
+Naming
+~~~~~~
+
+The root element and each other *important* element of the template should have a `name` attribute to ease the readability of the template.
+
+.. note::
+    The importance of an element in regard to this rule is left to the appreciation of the reviewer. If the structure of the template is hard to follow when reading just the first words of each line, it is probably a sign that some `name` attributes are missing on those lines.
+
+The name on the root element should follow the format `o-ComponentName`. And for children elements `o-ComponentName-partName` where:
+
+- `o-` is a must have Odoo prefix.
+- `ComponentName` is the PascalCase name of the current component/template.
+- `-` is a mandatory hyphen to separate the component name and the part name.
+- `partName` is the camelCase name of the child element referred by this `name` attribute.
+
+Attributes
+~~~~~~~~~~
+
+All the attributes of an element should be on the same line as the element itself. No character limit per line.
+
+.. note::
+    For extremely long lines, it is left to the appreciation of the reviewer to accept multi-line attributes within an element (but it is not mandatory).
+
+    In that case each attribute should be on its own line, with an indentation of 4 spaces relative to the element itself. The closing of the element (either `>` or `/>`) should be on its own line as well, with the same indentation as the corresponding opening character.
+
+The following order is applied for attributes:
```

Comment:
really??? I don't want to have to reread this page each time I review a template. this is extremely way too much.

Reply:
maybe ok as a suggested standard order, but not as an obligation.

Reply:
If we can agree on the guideline, then we can work on tools to validate it and apply it automatically.

The goal is not to have to remember the order by hand and waste time to enforce it manually.

If we can agree with the order on principle, that is all that matters for now I suppose.

Reply:
I am sorry, but I do not agree on the guideline. I do not see the additional value. Also, I do not want to add an extra compilation step for negligible value. I want to reduce friction, rather than increase it